be/src/vec/functions/url/domain.h

```diff
@@ -144,4 +145,138 @@ struct ExtractDomain {
     }
 };
 
+struct ExtractTopLevelDomain {
+    static size_t get_reserve_length_for_element() { return 5; }
+
+    static void execute(const char* data, size_t size, const char*& res_data, size_t& res_size) {
```

Comment:
why not return `StringRef`

Reply:
now the ExtractSubstringImpl function have abstract this function interface.
if changed maybe update other functions, it's a little more